crates/services/txpool_v2/src/storage/graph.rs

```diff
@@ -0,0 +1,568 @@
+    }
+
+    fn get_inner(&self, index: &NodeIndex) -> Result<&StorageData, Error> {
+        self.graph
+            .node_weight(*index)
+            .ok_or(Error::TransactionNotFound(format!(
+                "Transaction with index {:?} not found",
+                index
+            )))
+    }
+
+    fn get_dependents_inner(
+        &self,
+        index: NodeIndex,
+    ) -> Result<impl Iterator<Item = NodeIndex> + '_, Error> {
+        Ok(self
+            .graph
+            .neighbors_directed(index, petgraph::Direction::Outgoing))
+    }
+}
+
+impl Storage for GraphStorage {
+    type StorageIndex = NodeIndex;
+
+    fn store_transaction(
+        &mut self,
+        transaction: PoolTransaction,
+        dependencies: Vec<Self::StorageIndex>,
+        collided_transactions: &[Self::StorageIndex],
+    ) -> Result<(Self::StorageIndex, RemovedTransactions), Error> {
+        let tx_id = transaction.id();
+
+        // Add the new transaction to the graph and update the others in consequence
+        let tip = transaction.tip();
+        let gas = transaction.max_gas();
+        let outputs = transaction.outputs().clone();
+
+        // Check if the dependency chain is too big
+        let mut all_dependencies_recursively = HashSet::new();
+        let mut to_check = dependencies.clone();
+        while let Some(node_id) = to_check.pop() {
+            if collided_transactions.contains(&node_id) {
+                return Err(Error::Collided(
+                    "Use a collided transaction as a dependency".to_string(),
+                ));
+            }
+            // Already checked node
+            if all_dependencies_recursively.contains(&node_id) {
+                continue;
+            }
+            let Some(dependency_node) = self.graph.node_weight(node_id) else {
+                return Err(Error::Storage(format!(
+                    "Node with id {:?} not found",
+                    node_id
+                )));
+            };
+            if dependency_node.number_txs_in_chain >= self.config.max_txs_chain_count {
+                return Err(Error::NotInsertedChainDependencyTooBig);
+            }
+            all_dependencies_recursively.insert(node_id);
+            to_check.extend(self.get_dependencies(node_id)?);
+        }
+
+        if all_dependencies_recursively.len() >= self.config.max_txs_chain_count {
+            return Err(Error::NotInsertedChainDependencyTooBig);
+        }
+
+        // Remove collisions and their dependencies from the graph
+        let mut removed_transactions = vec![];
+        for collision in collided_transactions {
+            removed_transactions
+                .extend(self.remove_node_and_dependent_sub_graph(*collision)?);
+        }
+
+        let node = StorageData {
+            dependents_cumulative_tip: tip,
+            dependents_cumulative_gas: gas,
+            transaction,
+            number_txs_in_chain: all_dependencies_recursively.len().saturating_add(1),
```

Comment:
Mhhh we discussed this in https://github.com/FuelLabs/fuel-core/pull/2193. If I only have the children it will prevent us from catching dependency links with a lot of parents etc..

Reply:
`all_dependencies_recursively` already contains all parents. You can get too many parents only when you insert a new transaction. Removing of the transaction doesn't affect this rule. So by checking `all_dependencies_recursively.len() >= self.config.max_txs_chain_count` you cover the case with too many parents

Reply:
Ok, done.